### DISCUSSION TITLE
Mint.com data import

### DISCUSSION THREAD

**Discussion by BrianHumlicek** [TheAxelander/OpenBudgeteer#215] [General]
I am a former Mint.com user and want to import all of my historical transaction data.

For some background, Mint.com was a popular free online budgeting tool by Intuit.  As of December 2023, Intuit is discontinuing Mint.com and instead offering Credit Karma, which is awful.  As part of the transition, Mint.com is offering a download of your transaction data in .CSV format.  As Mint.com was a fairly large and popular offering, I think it would be worthwhile to include some baked in solution to import this data to make migration for these users relatively quick.  If the maintainers agree, then I would like to use this thread to discuss the way forward, and I can work creating a pull request.

So far, I have considered adding a tab or option to the existing Import page, but after some thought, I don't think this is the right place for it, as this is a one time data import, not a recurring task, which is what that page was designed to do.  Next, I considered leveraging the API, and creating a separate tool that would parse the data, and then load it through the API.  This would be fine for me, but then I began thinking there might be a lot of people that would like to do the same but aren't going to piece together tooling to do it.  So that brings me here to discuss the matter.  Do you think this is worth putting in the UI somewhere?

Brian

**Comment by TheAxelander** [comment]
Happy if you want to work on that. 
As this would be the first importer, I would propose to put it in a separate project like OpenBudgeteer.Importer and in Blazor I think it's fine to set up a separate page for it.

In `pre-release` I have reworked the navigation menu where I have now more space/flexibility. So I'm thinking of adding a (sub)menu for Importers and then adding this Mint.com importer as the first entry.

Can you maybe share already what kind of format (headers) this CSV file has?

**Comment by BrianHumlicek** [comment]
The data is nothing special.  The biggest deal is that it spans multiple accounts, so those need to be created along with the transactions.  Additionally it has a category field that I am trying to see if I can translate into buckets but I am not sure about that yet.

Here is a snippet of the csv.

```
"Date","Description","Original Description","Amount","Transaction Type","Category","Account Name","Labels","Notes"
"1/02/2024","BANK OF AMERICA, N.A.","BANK OF AMERICA, N.A.","18000.00","debit","Transfer","Capital One Savings","",""
"1/01/2024","BUC-EES","BUC-EES","28.48","debit","Gas & Fuel","Business Platinum Card®","",""
"1/01/2024","Wawa","WAWA GAS STORE","2.02","debit","Fast Food","Business Platinum Card®","",""
"1/01/2024","DETWILERS FARM MARKET","DETWILERS FARM MARKET","106.50","debit","Groceries","Business Platinum Card®","",""
"1/01/2024","BUC-EES","BUC-EES","23.13","debit","Gas & Fuel","Business Platinum Card®","",""
"1/01/2024","AplPay STARBUCKS XXX-XXX-7282 WA","AplPay STARBUCKS XXX-XXX-7282 WA","25.00","debit","Coffee Shops","Business Platinum Card®","",""
```

Brian

**Comment by TheAxelander** [comment]
Seems to be quite straight forward. Let me know if you need some help or input.
